From 51dd713d1b96545bdb4c3c5a5a330275c9760416 Mon Sep 17 00:00:00 2001
From: Luciana Fujii Pontello <luciana.fujii@collabora.co.uk>
Date: Sun, 25 Sep 2011 19:39:39 -0300
Subject: [PATCH 095/154] pvrvideosink: Handle delete event

When internal window is deleted, pvrvideosink handles it posting an
error message instead of being killed.
---
 sys/pvr2d/gstpvrvideosink.c |   11 +++++++++++
 1 files changed, 11 insertions(+), 0 deletions(-)

diff --git a/sys/pvr2d/gstpvrvideosink.c b/sys/pvr2d/gstpvrvideosink.c
index c0f5e9c..0a4cbff 100644
--- a/sys/pvr2d/gstpvrvideosink.c
+++ b/sys/pvr2d/gstpvrvideosink.c
@@ -390,6 +390,7 @@ gst_pvrvideosink_create_window (GstPVRVideoSink * pvrvideosink, gint width,
   GstXWindow *xwindow;
   GstDrawContext *dcontext;
   XGCValues values;
+  Atom wm_delete;
 
   GST_DEBUG_OBJECT (pvrvideosink, "begin");
 
@@ -411,6 +412,16 @@ gst_pvrvideosink_create_window (GstPVRVideoSink * pvrvideosink, gint width,
       width, height, 2, 2, pvrvideosink->dcontext->black);
   XSelectInput (dcontext->x_display, xwindow->window,
       ExposureMask | StructureNotifyMask);
+
+  /* Tell the window manager we'd like delete client messages instead of
+   * being killed */
+  wm_delete = XInternAtom (pvrvideosink->dcontext->x_display,
+      "WM_DELETE_WINDOW", True);
+  if (wm_delete != None) {
+    (void) XSetWMProtocols (pvrvideosink->dcontext->x_display, xwindow->window,
+        &wm_delete, 1);
+  }
+
   XMapWindow (dcontext->x_display, xwindow->window);
 
   /* We have to do that to prevent X from redrawing the background on
-- 
1.7.5.4

